packages/vm/src/runTx.ts

```diff
@@ -284,14 +284,15 @@ async function _runTx(this: VM, opts: RunTxOpts): Promise<RunTxResult> {
       // EIP-1559 spec:
       // The signer must be able to afford the transaction
       // `assert balance >= gas_limit * max_fee_per_gas`
-      const cost = tx.gasLimit.mul((tx as FeeMarketEIP1559Transaction).maxFeePerGas)
+      const cost = tx.gasLimit.mul((tx as FeeMarketEIP1559Transaction).maxFeePerGas).iadd(tx.value)
```

Comment:
can we just use `add` here instead of the in-place `iadd`? I want to make sure we don't mutate the original `tx.gasLimit` BN

Reply:
Sure! But this should return a copied BN though, since we `mul` and not `imul`.

Reply:
Ok sounds good I'll add a commit, yeah I figured but I think it's a better exercise of safety in case the line gets changed in the future. I'm sure it'd show up on some consensus tests though if it did get mutated 😄

Reply:
updated in https://github.com/ethereumjs/ethereumjs-monorepo/pull/1372/commits/03dc6d304ba59ca8c698860eb54a73b0e2b06b3d